File: tools/make_image_message.cpp
// tools/make_image_message.cpp
#include <iostream>
#include <cstring>
#include <filesystem>
#include "image_message_utils.hpp"

int main(int argc, char** argv) {
    const char* out = "data/image_message.bin";

    // Allow optional --out <path>
    for (int i = 1; i + 1 < argc; ++i) {
        if (std::strcmp(argv[i], "--out") == 0) {
            out = argv[i + 1];
        }
    }

    try {
        generate_image_message(out);
        
        // Calculate size for logging (4x4x1x1 float32 = 16 * 4 = 64 bytes data)
        size_t data_size = 4 * 4 * 1 * 1 * sizeof(float);
        std::cout << "Wrote " << out << " ("
                  << sizeof(ISMRMRD::ImageHeader) + data_size
                  << " bytes)\n";
        return 0;
    } catch (const std::exception& e) {
        std::cerr << "make_image_message error: " << e.what() << "\n";
        return 1;
    }
}

File: tools/make_image_message.py
# tools/make_image_message.py
# Usage: python3 tools/make_image_message.py [output_path]
# Default: image_message.bin in CWD
import sys
import numpy as np
import ismrmrd

out = sys.argv[1] if len(sys.argv) > 1 else "image_message.bin"

# Tiny 4x4 float32 image with ramp data
data = np.arange(16, dtype=np.float32).reshape(4, 4)
img = ismrmrd.Image.from_array(data)

# Minimal header fields
img.head.channels  = 1
img.head.data_type = ismrmrd.DATATYPE_FLOAT  # float32

with open(out, "wb") as f:
    f.write(img.getHead().tobytes())
    f.write(img.data.tobytes())

print(f"Wrote {out} ({len(img.getHead().tobytes()) + len(img.data.tobytes())} bytes)")
